### ISSUE TITLE
Warning when including javaFX in code

### ISSUE THREAD

**Issue by obrona** [nus-cs2103-AY2425S1/forum#136]
Thi is the warning message I am receiving.

Aug 30, 2024 4:15:58 PM com.sun.javafx.application.PlatformImpl startup
WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @2a62d1e2'
Aug 30, 2024 4:15:58 PM javafx.fxml.FXMLLoader$ValueElement processValue
WARNING: Loading FXML document with JavaFX API of version 20.0.1 by JavaFX runtime of version 17.0.7

How do I resolve this warning?

**Comment by damithc** [comment]
@obrona While we wait for others to pitch in with their inputs, you can carry on with the iP work (this warning doesn't need to be solved urgently).

**Comment by jinhanfromNUS** [comment]
Hi @obrona, I am a senior from your previous batch! 

According to the `WARNING: Loading FXML document with JavaFX API of version 20.0.1 by JavaFX runtime of version 17.0.7`, it seems that there is a version mismatch between the JavaFX API and the runtime version. Specifically, your code is using a JavaFX API from version 20.0.1, but it's being executed by a runtime that is version 17.0.7. 

Although I am not 100% certain that I provide the correct solution, I have some suggestions which might be able to help you.

**Suggestion 1: Ensure Consistent JavaFX Version:**
Update the JavaFX runtime to match the API version you're using. You can do this by downloading the appropriate JavaFX SDK and configuring your project to use it.
Please ensure your build.gradle file specifies the correct JavaFX version.

**Suggestion 2: Update Project Settings:**
Make sure your project is set to use the correct version of the JDK and JavaFX in your IDE settings. Check if the classpath is properly configured to point to the correct JavaFX SDK version.

However, as prof mentioned, I believe that you can still work on your current project first without any issue if both of the suggestions above do not work.

**Comment by ghost** [comment]
Hi, I am a senior from a previous batch

As jinhan mentioned, there is a mismatch between the JavaFX API that was being called from the FXML document (20.0.1) and the JavaFX version that was installed in your project (17.0.7)

I would assume that you have used SceneBuilder to create your FXML documents, as SceneBuilder will always use the most updated version of JavaFX when generating the FXML document, it will not always be in the same version as your JavaFX API that was included in your project

You could use any text editor to open your FXML document, and search for the following excerpt in your root node
```
xmlns="http://javafx.com/javafx/20.0.1"
```
And replace it with
```
xmlns="http://javafx.com/javafx/17.0.7"
```
This will ensure that the FXML documents only call API that is available for JavaFX 17.0.7, and the warning should stop appearing

**Comment by obrona** [comment]
I have resolved the 2nd error
```
Aug 30, 2024 4:15:58 PM javafx.fxml.FXMLLoader$ValueElement processValue
WARNING: Loading FXML document with JavaFX API of version 20.0.1 by JavaFX runtime of version 17.0.7
```

But I still have the 1st error
```
Aug 31, 2024 10:08:02 AM com.sun.javafx.application.PlatformImpl startup
WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @58e7b29e'
```

How do I resolve this ?

**Comment by ghost** [comment]
The TL;DR is that the setup that you were taught to use in CS2103 does not import JavaFX as a named module, but rather only include it as a dependency class to be loaded in your main app, and this is not the supported installation configuration for JavaFX 9 and above. This warning should not impact your application and you could ignore it as is. If you're looking to resolve this, you can take a look at the following links

https://www.oracle.com/sg/corporate/features/understanding-java-9-modules.html
https://stackoverflow.com/questions/67854139/javafx-warning-unsupported-javafx-configuration-classes-were-loaded-from-unna

**Comment by damithc** [comment]
I'm adding the following (from the JavaFX tutorial part 1), in case someone else read this thread later:

![image](https://github.com/user-attachments/assets/e91b96ba-5f1e-4ff8-a610-a5260ed6fe0f)
